src/braket/devices/local_simulator.py

```diff
@@ -295,6 +301,26 @@ def _(
         results = simulator.run(program, shots, *args, **kwargs)
         return GateModelQuantumTaskResult.from_object(results)
 
+    @_run_internal.register
+    def _(
+        self,
+        program: aq.Program,
+        shots: Optional[int] = None,
+        inputs: Optional[Dict[str, float]] = None,
+        *args,
+        **kwargs,
+    ):
+        simulator = self._delegate
+        if DeviceActionType.OPENQASM not in simulator.properties.action:
+            raise NotImplementedError(f"{type(simulator)} does not support OpenQASM programs")
+        program = Program(source=program.to_ir(ir_type=IRType.OPENQASM))
+        results = simulator.run(program, shots, *args, **kwargs)
```

Comment:
If we make this change, we can remove the bit about `mcm=True` from the README (and also from the tests, if we want).
```suggestion
        program = Program(source=program.to_ir(ir_type=IRType.OPENQASM))
        kwargs["mcm"] = True
        results = simulator.run(program, shots, *args, **kwargs)
```

Reply:
Yes I like this, since I'm not convinced `mcm=True` is the final interface here, so I'd prefer if we could just have it one place